### PR TITLE
Additional docs on external and master job cache, assorted doc fixes

### PR DESCRIPTION
**Pull request by jacobhammons**
Refs #25277

### CODE REVIEW

File: doc/topics/jobs/external_cache.rst [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+:ref:`Default Job Cache <default_job_cache>`.
+
+In addition to the Default Job Cache, Salt provides two additional
+mechanisms to send job results to other systems (databases, local syslog,
+and others):
+
+* External Job Cache
+* Master Job Cache 
+
+The major different between these two
```
Comment (rallytime) [inline doc/topics/jobs/external_cache.rst]: difference?

File: doc/topics/jobs/external_cache.rst [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+* Advantages: Data is stored without placing additional load on the Salt Master.
+* Disadvantages: Each Salt Minion connects to the external job cache, which can
+  result in a large number of connections.  Also requires additional configuration to
+  get returner module settings on all Salt Minions.
+
+Master Job Cache - Master-Side Returner
+---------------------------------------
+
+.. versionadded:: 2014.7
```
Comment (rallytime) [inline doc/topics/jobs/external_cache.rst]: 2014.7.0

**Comment by rallytime** [comment]
I like those images and the new doc. Super helpful and clear! +1

**Comment by bechtoldt** [comment]
this needs a :+1: :)

**Comment by jacobhammons** [comment]
@rallytime @bechtoldt Thanks, I appreciate the reviews and feedback!

**Pull request merged**